filebeat/input/log/harvester.go

```diff
@@ -52,6 +52,11 @@ var (
 	ErrClosed       = errors.New("reader closed")
 )
 
+// OutletFactory provides an outlet for the harvester
+type OutletFactory interface {
+	GetOutlet() channel.Outleter
+}
```

Comment:
This type can be changed to `type OutletFactory func() channel.Outleter`.

By just passing a function/closure, we don't need to add another public method to each Input type, but still get the lazy initialization we want.